zoo/api/query.py

```diff
@@ -142,3 +153,80 @@ def resolve_all_dependency_usages(self, info, **kwargs):
         return types.DependencyUsageConnection(
             page_info=page_info, edges=edges, total_count=total
         )
+
+    def resolve_all_search_results(self, info, **kwargs):
+        paginator = Paginator(**kwargs)
+        query_param = kwargs["search_query"] if "search_query" in kwargs else ""
+        search_type = (
+            kwargs["search_type"]
+            if "search_type" in kwargs
+            else IndexType.Service.value
+        )
+
+        indexes = meili_client.get_indexes()
+        # get all total counts for each meili index
+        search_index = None
+        total_count = {}
+        for index in indexes:
+            total_count[index["uid"]] = meili_client.get_index(index["uid"]).search(
+                query=query_param, opt_params={"offset": 0, "limit": 1}
+            )["nbHits"]
+
+            if index["uid"] == search_type:
+                search_index = index
+
+        # get all results for requested index
+        search_results = Query._get_all_for_index(
+            query_param, search_index, total_count[search_index["uid"]]
+        )
+        edges = []
+        for i, search_result in enumerate(
+            search_results[paginator.slice_from : paginator.slice_to]
+        ):
+            cursor = paginator.get_edge_cursor(i + 1)
+            edges.append(
+                types.SearchResultsConnection.Edge(node=search_result, cursor=cursor)
+            )
+
+        return types.SearchResultsConnection(
+            page_info=paginator.get_page_info(total_count[search_type]),
+            edges=edges,
+            total_services_count=total_count[IndexType.Service.value],
+            total_analytics_count=total_count[IndexType.Dependency.value],
+        )
+
+    @staticmethod
+    def _get_all_for_index(query_param, search_index, total_count):
+        offset = 0
+        limit = 100
+        search_results = []
+        while total_count != len(search_results):
+            results = meili_client.get_index(search_index["uid"]).search(
+                query=query_param, opt_params={"offset": offset, "limit": limit}
+            )
+
+            search_results = Query._objects_from_result(
```

Comment:
this should be appending, right?

Reply:
oh, I see now... it's modifying the `search_results` variable inside the function 😢 

this is a no-no for me